nexus/reconfigurator/planning/src/mgs_updates/mod.rs

```diff
@@ -476,6 +652,179 @@ fn try_make_update_sp(
+            baseboard_id
+        );
+        return None;
+    };
+
+    let active_slot = rot_state.active_slot;
+
+    let active_caboose = match active_slot {
+        RotSlot::A => CabooseWhich::RotSlotA,
+        RotSlot::B => CabooseWhich::RotSlotB,
+    };
+
+    let Some(active_caboose) =
+        inventory.caboose_for(active_caboose, baseboard_id)
+    else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing active slot {active_slot} caboose from inventory)";
+            baseboard_id,
+        );
+        return None;
+    };
+
+    let Ok(expected_active_version) = active_caboose.caboose.version.parse()
+    else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (cannot parse current active version as an ArtifactVersion)";
+            baseboard_id,
+            "found_version" => &active_caboose.caboose.version,
+        );
+        return None;
+    };
+
+    let board = &active_caboose.caboose.board;
+    let matching_artifacts: Vec<_> = current_artifacts
+        .artifacts
+        .iter()
+        .filter(|a| {
+            // A matching RoT artifact will have:
+            //
+            // - "name" matching the board name (found above from caboose)
+            // - "kind" matching one of the known RoT kinds
+
+            if a.id.name != *board {
+                return false;
+            }
+
+            match active_slot {
+                RotSlot::A => {
+                    let slot_a_artifacts = [
+                        ArtifactKind::GIMLET_ROT_IMAGE_A,
+                        ArtifactKind::PSC_ROT_IMAGE_A,
+                        ArtifactKind::SWITCH_ROT_IMAGE_A,
+                    ];
+
+                    if slot_a_artifacts.contains(&a.id.kind) {
+                        return true;
+                    }
+                }
+                RotSlot::B => {
+                    let slot_b_artifacts = [
+                        ArtifactKind::GIMLET_ROT_IMAGE_B,
+                        ArtifactKind::PSC_ROT_IMAGE_B,
+                        ArtifactKind::SWITCH_ROT_IMAGE_B,
+                    ];
+
+                    if slot_b_artifacts.contains(&a.id.kind) {
+                        return true;
+                    }
+                }
+            }
+
+            false
+        })
+        .collect();
+    if matching_artifacts.is_empty() {
+        warn!(
+            log,
+            "cannot configure RoT update for board (no matching artifact)";
+            baseboard_id,
+        );
+        return None;
+    }
+
+    if matching_artifacts.len() > 1 {
+        // This should be impossible unless we shipped a TUF repo with more
+        // than 1 artifact for the same board and slot. But it doesn't prevent
+        // us from picking one and proceeding. Make a note and proceed.
+        warn!(log, "found more than one matching artifact for RoT update");
```

Comment:
My comment about the signature makes me wonder if we should make this an error instead to catch that kind of problem.